src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [chosenTodo, setChosenTodo] = useState<Todo | null>(null);
+  const [searchInput, setSearchInput] = useState<string>('');
+  const [filter, setFilter] = useState<string>('all');
+  const [chosenUser, setChosenUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(
+        setTodos,
+      );
+
+    setIsLoading(false);
+  }, []);
+
+  useEffect(() => {
+    if (chosenTodo !== null) {
+      getUser(chosenTodo.userId)
+        .then(
+          setChosenUser,
+        );
+    }
+  }, [chosenTodo]);
+
+  const handleFilterTodos = () => {
+    let todosfiltered = [...todos];
+
+    if (searchInput !== '') {
+      todosfiltered = todosfiltered.filter((todo) => (todo.title.toLocaleLowerCase().includes(searchInput.toLocaleLowerCase())));
+    }
+
+    if (filter === 'active') {
+      todosfiltered = todosfiltered.filter((todo) => todo.completed === false);
+    }
+
+    if (filter === 'completed') {
+      todosfiltered = todosfiltered.filter((todo) => todo.completed === true);
+    }
+
+    return todosfiltered;
+  };
```

Comment:
please consider handling todo filtering outside of the component I don't think it needs to be there, and it makes your component bigger and bigger